specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2023-08-01-preview/mfe.json

```diff
@@ -24574,6 +24641,20 @@
       "x-ms-client-name": "ServerlessEndpointProperties",
       "additionalProperties": false
     },
+    "ServerlessEndpointStatus": {
+      "type": "object",
+      "properties": {
+        "metrics": {
+          "type": "object",
+          "additionalProperties": {
```

Comment:
>additionalProperties [](http://example.com/codeflow?start=11&length=20)

Additional properties pattern is an ARM anti-pattern. 
Generally, this pattern should only be used when needed and the object properties are dynamic, unknown, or user defined. Ref: [https://armwiki.azurewebsites.net/api_contracts/guidelines/openapi.html#oapi032-only-use-additionalproperties-when-the-object-properties-are-dynamic-unknown-or-user-defined](https://armwiki.azurewebsites.net/api_contracts/guidelines/openapi.html#oapi032-only-use-additionalproperties-when-the-object-properties-are-dynamic-unknown-or-user-defined).

Can you schematize this?

Reply:
In this case, the information dynamic. The list of metrics will be model dependent, some of which may come from 3P ISVs etc. Hence it is modeled as a generic string dictionary.

Reply:
Please book an ARM modeling office hours slot and go over the design with the ARM API reviewer to assess if this pattern can be approved.

Reply:
We have booked an office hour, however we are concerned about the time delay so wondering if we can close on this faster over IM or a team call.

The key thing here is that our service is not in control of all possible keys in this dictionary, because this data comes from customers (specifically, model providers). We are hoping this is a valid use case of additional properties